### PR TITLE
add --build-on-remote flag

### PR DESCRIPTION
**Pull request by Lassulus**

### CODE REVIEW

File: src/nixos-anywhere.sh [hunk 1]
```diff
@@ -173,8 +178,10 @@ if [[ -n ${flake-} ]]; then
     echo 'For example, to use the output nixosConfigurations.foo from the flake.nix, append "#foo" to the flake-uri.' >&2
     exit 1
   fi
-  disko_script=$(nix_build "${flake}#nixosConfigurations.\"${flakeAttr}\".config.system.build.disko")
-  nixos_system=$(nix_build "${flake}#nixosConfigurations.\"${flakeAttr}\".config.system.build.toplevel")
+  if [[ ${build_on_remote-n} == "n" ]]; then
+    disko_script=$(nix_build "${flake}#nixosConfigurations.\"${flakeAttr}\".config.system.build.disko")
+    nixos_system=$(nix_build "${flake}#nixosConfigurations.\"${flakeAttr}\".config.system.build.toplevel")
```
Comment (Mic92) [inline src/nixos-anywhere.sh:183]:
```suggestion
    nixos_system=$(nix_build "${flake}#nixosConfigurations.\"${flakeAttr}\".config.system.build.toplevel")
    elif [[ -z "${flake}" ]]; then
      abort "--build-on-remote also requires --flake to be set"
```

Comment (Lassulus) [reply src/nixos-anywhere.sh:183]: well it doesn't need flakes. but it has no effect without flakes, since the paths will already be build

Comment (Mic92) [reply src/nixos-anywhere.sh:183]: if there are no flakes how does it work? I don't see where machines are build on the remote?

Comment (Mic92) [reply src/nixos-anywhere.sh:183]: Ah. I right I thought we also supported non-flake options but we actually do not for nixos-anywhere.

**Comment by Mic92** [comment]
bors merge

**Review by Mic92 — APPROVED** [review]

**Pull request merged**